### PR TITLE
fix to nsrt learning unify

### PR DESCRIPTION
**Pull request by tomsilver**
in the case where options are known, we should use the parameterizedoption (and not just the option vars) when unifying in nsrt learning.

### CODE REVIEW

File: src/nsrt_learning.py [hunk 1]
```diff
@@ -234,13 +243,18 @@ def unify_effects_and_options(
...
         ground_option_args: Tuple[Object, ...],
-        lifted_option_args: Tuple[Variable, ...],
+        lifted_option_args: Tuple[Variable, ...]
 ) -> Tuple[bool, ObjToVarSub]:
     """Wrapper around utils.unify() that handles option arguments, add effects,
     and delete effects. Changes predicate names so that all are treated
     differently by utils.unify().
     """
+    # Can't unify if the parameterized options are different.
+    if ground_param_option != lifted_param_option:
```
Comment (ronuchit) [inline src/nsrt_learning.py:256]: Note, of course, we could directly check this in the loop above. But fine to keep all the unification logic in one place, even if it's trivial as in this case.

File: tests/test_nsrt_learning.py [hunk 2]
```diff
@@ -319,7 +322,26 @@ def test_unify_effects_and_options():
...
         lifted_option_args)
     assert not suc
     assert not sub
+    # The following test is for an edge case where everything is identical
+    # except for the name of the parameterized option. We do not want to
+    # unify in this case.
+    param_option1 = ParameterizedOption(
+        "dummy1", [cup_type], Box(0.1, 1, (1,)), lambda s, m, o, p: Action(p),
+        lambda s, m, o, p: False, lambda s, m, o, p: False)
+    suc, sub = unify_effects_and_options(
```
Comment (ronuchit) [inline tests/test_nsrt_learning.py:337]: To hammer home this test, let's also include a copy-paste of it where you pass in param_option0 twice, and all the other arguments are the same -- in this case, `suc` should be True

**Comment by ronuchit** [comment]
Thank you for catching this!
